File: old-python/ProcessData.py
import pandas as pd

#because its a .csv file, first we are gonna convert our dataset to a txt file, removing some of the headers and stuff

#this funct will handle the data, for our main Poem Generator
def HandleData():
    #first lets open our file

    # p = pd.read_csv("poetry.csv", usecols=['content'])

    # print(p)

    ##each row is a poem, so lets conver it into a txt file, so we can work with it easier

    #for each poem in the content column, we add it to the file
    with open("poems.txt", "w", encoding="utf-8") as f:
        for poem in p['content']:
            #add the poem, and then make a new line
            f.write(poem + "\n")


    #a list of chars that we dont want
    remove_chars = r"""/\.,:;'+=-|"'!@#$%^&*())1234567890"""
    #this will also handle the removing unwanted punctuation and chars:
    #we use with open, as it handles closing the file for us
    finallines = ""
    with open("poems.txt", "r", encoding="utf-8") as f:
        #ok, for each line, in the file, lets remove the "/\.,:;'+=-| and whatever other chars there are
        for line in f:
            #the way it works, is it takes in a line, and attempts to translate an chars to whatever i set. we use this instead of replace
            #because it lets me do multiple making a transition table.
            cleanline = line.translate(str.maketrans("", "", remove_chars))
            finallines += cleanline;



        
    print(finallines)

    #the W mode, removes all text on open
    with open("poems.txt", "w") as file:
        file.write(finallines)


    print("Done!")

File: C++AI/C++AI/ProcessData.cpp
#include "ProcessData.h"
#include <fstream> //for loading files
#include <iostream> //for printing text, and calling some functs
#include <string> //for handling strings
#include <sstream> //for our buffer

//this funct takes our data, and handles it!
//its hard to handle something like a csv, so we are just gonna handle the txt side
int handleData()
{

	//we need this for our write
	std::string poemText;



	//ok open the file!
	std::fstream poem("poems.txt");

	//then check we got it
	if(!poem.is_open())
	{ 
		std::cout << "ERROR, COULD NOT OPEN FILE " << std::endl;
		return -1;
	}
	else {
		//ok we made sure that its open!, so we are gonna parse our file
		//what the plan is, we have a var this is = to the poem val, then we get the amount of chars
		//then go through each char, compare to our list of unwanted chars, and remove them! 
		//Then write to the file
		//then save it!

		//we can load everything at once, using a buffer
		std::stringstream buffer;

		buffer << poem.rdbuf();

		//then send it to a string

		poemText = buffer.str();

		//make a list of our unwanted chars

		std::string remove_chars = "/\\.,:;'+=-|\"'!@#?[]{}$%^&*())1234567890";

		//loop backwards
		std::string cleanedText;
		cleanedText.reserve(poemText.size());
		for (char c : poemText)
		{
			//ok, now that we loop through each char, lets check
			//super easy, like my c++ parser!
			//we check to see if we can find a char, that is the same in remove_chars
			//then erase!
			if (remove_chars.find(c) == std::string::npos)
			{
				//ok, we have found the char
				//erase is slow, but idrc, as this will run once!
				cleanedText.push_back(c);
			}
		}
		poemText.swap(cleanedText);

		//teaches the ai new lines!
		size_t pos = 0;
		//we need to teach it <newline tags, so we find the \n, and if we find one, we replace this, we go through the list!
		//this is super easy, as this is normal in the browser engine!
		while ((pos = poemText.find('\n', pos)) != std::string::npos)
		{
			poemText.replace(pos, 1, " <NEWLINE> ");
			pos += 11; //after we change it, move past to avoid gliches
		}

		//close before we write
		poem.close();


		//ok we are done, lets write!
		std::ofstream outPoem("poems.txt");
		if (!outPoem.is_open())
		{
			std::cout << "ERROR, COULD NOT OPEN FILE " << std::endl;
			return -1;
		}

		//done!
		outPoem << poemText;

		outPoem.close();

		std::cout << "Done!" << std::endl;

	}
	

	return 0;
}